documentation/docs/30-advanced/30-link-options.md

```diff
@@ -6,20 +6,55 @@ In SvelteKit, `<a>` elements (rather than framework-specific `<Link>` components
+Before the browser registers that the user has clicked on a link, we can detect that they've hovered the mouse over it (on desktop) or that a `touchstart` or `mousedown` event has happened. In both cases, we can make an educated guess that a `click` event is coming.
 
-To apply this behaviour across the board, add the attribute to a parent element (or even the `<body>` in your `src/app.html`):
+SvelteKit can use this information to get a head start on importing the code and fetching the page's data, which can give us an extra couple of hundred milliseconds — the difference between a user interface that feels laggy and one that feels snappy.
+
+We can control this behaviour with the `data-sveltekit-preload-data` attribute, which can have one of two values:
+
+- `"hover"` means that preloading will start if the mouse comes to a rest over a link
+- `"tap"` means that preloading will start as soon as a `touchstart` or `mousedown` event is registered
+
+> On mobile, links with `"hover"` will preload on `touchstart`.
+
+The default project template has a `data-sveltekit-preload-data="hover"` attribute applied to the `<body>` element in `src/app.html`, meaning that every link is preloaded on hover by default:
+
+```html
+<body data-sveltekit-preload-data="hover">
+	<div style="display: contents">%sveltekit.body%</div>
+</body>
+```
+
+Sometimes, calling `load` when the user hovers over a link might be undesirable, either because it's likely to result in false positives (a click needn't follow a hover) or because data is updating very quickly and a delay could mean staleness.
+
+In these cases, you can specify the `"tap"` value, which causes SvelteKit to call `load` only when the user taps or clicks on a link:
 
 ```html
-/// file: src/routes/+layout.svelte
-<main data-sveltekit-prefetch>
-	<slot />
-</main>
+<a data-sveltekit-preload-data="tap" href="/stonks">
+	Get current stonk values
+</a>
 ```
 
-> You can also programmatically invoke `prefetch` from `$app/navigation`.
+> You can also programmatically invoke `preloadData` from `$app/navigation`.
+
+Data will never be preloaded if the user has chosen reduced data usage, meaning [`navigator.connection.saveData`](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) is `true`.
+
+### data-sveltekit-preload-code
+
+Even in cases where you don't want to preload _data_ for a link, it can be beneficial to preload the _code_. The `data-sveltekit-preload-code` attribute works similarly to `data-sveltekit-preload-data`, except that it can take one of four values, in decreasing 'eagerness':
+
+- `"page"` means that all links on the page will be preloaded
```

Comment:
maybe call it `links` or `all`?

Reply:
`links` doesn't make sense to me because every value is about links. `all` makes me ask 'all what?' `page` does feel like an outlier though, since the others all describe events (if you allow `viewport` to mean 'enter viewport'), so perhaps it's not ideal. i'm not sure what else it could be though. i wondered about `navigate` but that doesn't really make sense because the event precedes the creation of the link.

some other ideas:

* `data-sveltekit-preload-code="now"`
* `data-sveltekit-preload-code="eager"`
* `data-sveltekit-preload-code="immediate"`

Reply:
went for `eager`